lib/ws-kernel-picker.js

```diff
@@ -56,23 +63,29 @@ class CustomListView {
 }
 
 export default class WSKernelPicker {
-  constructor(onChosen) {
+  _onChosen: (kernel: Kernel) => void;
+  _kernelSpecFilter: (kernelSpec: Kernelspec) => boolean;
+  _path: string;
+  previouslyFocusedElement: ?HTMLElement;
+
+  constructor(onChosen: (kernel: Kernel) => void) {
     this._onChosen = onChosen;
   }
 
-  toggle(_grammar, _kernelSpecFilter) {
-    this._grammar = _grammar;
+  toggle(_kernelSpecFilter: (kernelSpec: Kernelspec) => boolean) {
     this._kernelSpecFilter = _kernelSpecFilter;
-    const gateways = Config.getJson("gateways", []);
+    const gateways = Config.getJson("gateways") || [];
     if (_.isEmpty(gateways)) {
       atom.notifications.addError("No remote kernel gateways available", {
         description:
           "Use the Hydrogen package settings to specify the list of remote servers. Hydrogen can use remote kernels on either a Jupyter Kernel Gateway or Jupyter notebook server."
       });
       return;
     }
+    const path = store.editor ? store.editor.getPath() : null;
+    if (!path) return;
```

Comment:
What should be the expected behavior when running `hydrogen:connect-to-remote-kernel` with an unsaved editor? 

I figure we would use the project path? An empty string? For now just returning to avoid a flow error.